### PR TITLE
Add DynamicKuboToyabe fitting function

### PR DESCRIPTION
**Pull request by raquelalvarezbanos**
Fixes trac issue #9556

For tester: Open the muon analysis interface and load one of the muon datasets attached in the trac ticket. Set Alpha=0.96 in the first row in "Pair Table", in "Grouping Options" tab, and hit "Plot". Go to "Data Analysis" and add "DynamicKuboToyabe" fitting function. You should be able to obtain a good fit if you fix the Field parameter to 0 (otherwise an exception will be raised, as the DKT function for non-zero magnetic field will be implemented in ticket #11086). Play with different fitting ranges and the other dataset attached.

### CODE REVIEW

File: Code/Mantid/Framework/CurveFitting/inc/MantidCurveFitting/DynamicKuboToyabe.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+//#include "MantidAPI/IFunctionMW.h"
+//#include "MantidAPI/IFunction1D.h"
+
+namespace Mantid
+{
+  namespace CurveFitting
+  {
+    /** 
+     Provide Dynamic Kubo Toyabe function interface to IPeakFunction for muon scientists.
```
Comment (NickDraper) [inline Code/Mantid/Framework/CurveFitting/inc/MantidCurveFitting/DynamicKuboToyabe.h]: This isn't an IPeakFunction, could you correct the comment

File: Code/Mantid/Framework/CurveFitting/inc/MantidCurveFitting/DynamicKuboToyabe.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+   
+     @author Karl Palmen, ISIS, RAL
```
Comment (NickDraper) [inline Code/Mantid/Framework/CurveFitting/inc/MantidCurveFitting/DynamicKuboToyabe.h]: and add your name here

File: Code/Mantid/Framework/CurveFitting/src/DynamicKuboToyabe.cpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+{
+namespace CurveFitting
+{
+
+using namespace Kernel;
+using namespace API;
+
+DECLARE_FUNCTION(DynamicKuboToyabe)
+
+// ** MODIFY THIS **
```
Comment (NickDraper) [inline Code/Mantid/Framework/CurveFitting/src/DynamicKuboToyabe.cpp]: These comments need removing

File: Code/Mantid/Framework/CurveFitting/src/DynamicKuboToyabe.cpp [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+// Static Zero Field Kubo Toyabe relaxation function
+double ZFKT (const double x, const double G){
+
+  const double q = G*G*x*x;
+  return (0.3333333333 + 0.6666666667*exp(-0.5*q)*(1-q));
+}
+
+// Static Non-Zero field Kubo Toyabe relaxation function
```
Comment (NickDraper) [inline Code/Mantid/Framework/CurveFitting/src/DynamicKuboToyabe.cpp]: Take away the dead code

File: Code/Mantid/Framework/CurveFitting/src/DynamicKuboToyabe.cpp [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+//{
+//  throw std::runtime_error("HKT not implemented yet");
+//}
+
+// Dynamic Kubo-Toyabe
+double getDKT (double t, double G, double v){
+
+  const int tsmax = 656; // Length of the time axis, 32 us of valid data
+  const double eps = 0.05; // Bin width for calculations
+  //  const int stk = 25; // Not used for the moment
```
Comment (NickDraper) [inline Code/Mantid/Framework/CurveFitting/src/DynamicKuboToyabe.cpp]: Unless there is a good reason for keeping this it should be removed

File: Code/Mantid/Framework/CurveFitting/src/DynamicKuboToyabe.cpp [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+void DynamicKuboToyabe::function1D(double* out, const double* xValues, const size_t nData)const
+{
+  const double& A = getParameter("Asym");
+  const double& G = fabs(getParameter("Delta"));
+  const double& F = fabs(getParameter("Field"));
+  const double& v = fabs(getParameter("Nu"));
+
+
+    // Zero hopping rate
+	if (v == 0.0) {
```
Comment (NickDraper) [inline Code/Mantid/Framework/CurveFitting/src/DynamicKuboToyabe.cpp]: very confusing indentation here, could you try running clang on this? or fix it yourself

**Comment by NickDraper** [comment]
Functionally fine, but needs some neatening up in the code

File: Code/Mantid/Framework/CurveFitting/src/DynamicKuboToyabe.cpp (hunk 5 above)
Comment (raquelalvarezbanos) [inline Code/Mantid/Framework/CurveFitting/src/DynamicKuboToyabe.cpp]: The only reason for keeping it is that the HKT function will be used in the future (trac issue #11086), once I know how to implement it (there are some details that need to be discussed with muon scientists). But it is true that it is not used now, so I will be removing it for this ticket.

**Comment by raquelalvarezbanos** [comment]
Thanks for your comments, Nick. I will update the code according to your suggestions.

**Pull request merged**